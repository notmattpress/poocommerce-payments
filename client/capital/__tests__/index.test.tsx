/**
 * External dependencies
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
/**
 * Internal dependencies
 */
import { useActiveLoanSummary, useLoans } from 'wcpay/data';
import CapitalPage from '..';

// Mock the useLoans hook
jest.mock( 'wcpay/data', () => ( {
	useLoans: jest.fn(),
	useActiveLoanSummary: jest.fn(),
} ) );

jest.mock( '@poocommerce/data', () => ( {
	useUserPreferences: jest.fn( () => ( {
		updateUserPreferences: jest.fn(),
		wc_payments_wporg_review_2025_prompt_dismissed: false,
	} ) ),
} ) );

declare const global: {
	wcpaySettings: {
		zeroDecimalCurrencies: string[];
		testMode: boolean;
		connect: {
			country: string;
		};
		accountLoans: {
			has_active_loan: boolean;
		};
		dateFormat: string;
	};
};

describe( 'CapitalPage', () => {
	beforeEach( () => {
		global.wcpaySettings = {
			zeroDecimalCurrencies: [],
			connect: {
				country: 'US',
			},
			accountLoans: { has_active_loan: true },
			testMode: true,
			dateFormat: 'M j, Y',
		};
	} );

	it( 'renders the TableCard component with loan data', () => {
		const loans = [
			{
				wcpay_capital_loan_id: 1,
				stripe_account_id: 'acct_1QD3WwR45o7u2iE3',
				stripe_offer_id: 'financingoffer_1QEpQ2J5cIRIG92xWoVJJE3S',
				stripe_loan_id: 'financingoffer_1QEpQ2J5cIRIG92xWoVJJE3S',
				amount: 100000,
				currency: 'usd',
				fee_amount: 10000,
				withhold_rate: 0.15,
				accepted_at: '2024-10-28 10:10:50',
				paid_out_at: '2024-10-28 10:11:40',
				first_paydown_at: null,
				fully_paid_at: null,
				deleted_at: null,
			},
		];
		( useLoans as jest.Mock ).mockReturnValue( {
			loans,
			isLoading: false,
		} );

		( useActiveLoanSummary as jest.Mock ).mockReturnValue( {
			summary: {
				details: {
					advance_amount: 100000,
					advance_paid_out_at: 1729505500.7316985,
					currency: 'usd',
					current_repayment_interval: {
						due_at: 1735294500,
						paid_amount: 0,
						remaining_amount: 12223,
					},
					fee_amount: 10000,
					paid_amount: 0,
					remaining_amount: 110000,
					repayments_begin_at: 1730110500,
					withhold_rate: 0.15,
				},
				isLoading: false,
			},
		} );

		const { container } = render( <CapitalPage /> );

		expect(
			screen.getByRole( 'heading', { name: 'All loans' } )
		).toBeInTheDocument();
		expect( container ).toMatchSnapshot();
	} );

	it( 'shows loading state when loans are being fetched', () => {
		( useLoans as jest.Mock ).mockReturnValue( {
			loans: [],
			isLoading: true,
		} );

		render( <CapitalPage /> );

		expect(
			screen.getByText( 'Your requested data is loading' )
		).toBeInTheDocument();
	} );
} );
