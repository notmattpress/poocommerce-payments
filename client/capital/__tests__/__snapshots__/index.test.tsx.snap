// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CapitalPage renders the TableCard component with loan data 1`] = `
.emotion-0 {
  background-color: #fff;
  color: #1e1e1e;
  position: relative;
  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.1);
  outline: none;
  border-radius: calc(8px - 1px);
}

.emotion-2 {
  height: 100%;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: calc(4px * 2);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  border-bottom: 1px solid;
  box-sizing: border-box;
  border-color: rgba(0, 0, 0, 0.1);
  padding: calc(4px * 4) calc(4px * 6);
}

.emotion-4>* {
  min-width: 0;
}

.emotion-4:last-child {
  border-bottom: none;
}

.emotion-4:first-of-type {
  border-top-left-radius: calc(8px - 1px);
  border-top-right-radius: calc(8px - 1px);
}

.emotion-4:last-of-type {
  border-bottom-left-radius: calc(8px - 1px);
  border-bottom-right-radius: calc(8px - 1px);
}

.emotion-6 {
  display: block;
  max-height: 100%;
  max-width: 100%;
  min-height: 0;
  min-width: 0;
}

.emotion-10 {
  box-sizing: border-box;
  height: auto;
  max-height: 100%;
  padding: calc(4px * 4) calc(4px * 6);
}

.emotion-10:first-of-type {
  border-top-left-radius: calc(8px - 1px);
  border-top-right-radius: calc(8px - 1px);
}

.emotion-10:last-of-type {
  border-bottom-left-radius: calc(8px - 1px);
  border-bottom-right-radius: calc(8px - 1px);
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: calc(4px * 2);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.emotion-12>* {
  min-width: 0;
}

.emotion-14 {
  display: block;
  max-height: 100%;
  max-width: 100%;
  min-height: 0;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-30 {
  background: transparent;
  display: block;
  margin: 0!important;
  pointer-events: none;
  position: absolute;
  will-change: box-shadow;
  border-radius: inherit;
  bottom: 0;
  box-shadow: 0 0px 0px 0 rgba(0, 0, 0, 0);
  opacity: 1;
  left: 0;
  right: 0;
  top: 0;
  border-radius: 8px;
}

@media not ( prefers-reduced-motion ) {
  .emotion-30 {
    -webkit-transition: box-shadow 200ms cubic-bezier(0.08, 0.52, 0.52, 1);
    transition: box-shadow 200ms cubic-bezier(0.08, 0.52, 0.52, 1);
  }
}

.emotion-34 {
  background-color: #fff;
  color: #1e1e1e;
  position: relative;
  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.1);
  outline: none;
  border-radius: calc(2px - 1px);
}

.emotion-36 {
  height: 100%;
}

.emotion-38 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  border-bottom: 1px solid;
  box-sizing: border-box;
  border-color: rgba(0, 0, 0, 0.1);
  padding: calc(4px * 4) calc(4px * 6);
}

.emotion-38>*+*:not( marquee ) {
  margin-left: calc(4px * 2);
}

.emotion-38>* {
  min-width: 0;
}

.emotion-38:last-child {
  border-bottom: none;
}

.emotion-38:first-of-type {
  border-top-left-radius: calc(2px - 1px);
  border-top-right-radius: calc(2px - 1px);
}

.emotion-38:last-of-type {
  border-bottom-left-radius: calc(2px - 1px);
  border-bottom-right-radius: calc(2px - 1px);
}

.emotion-40 {
  color: #1e1e1e;
  line-height: 1.2;
  margin: 0;
  color: #23282d;
  font-size: calc((16 / 13) * 13px);
  font-weight: 600;
}

.emotion-42 {
  box-sizing: border-box;
  height: auto;
  max-height: 100%;
}

.emotion-42:first-of-type {
  border-top-left-radius: calc(2px - 1px);
  border-top-right-radius: calc(2px - 1px);
}

.emotion-42:last-of-type {
  border-bottom-left-radius: calc(2px - 1px);
  border-bottom-right-radius: calc(2px - 1px);
}

.emotion-44 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  border-top: 1px solid;
  box-sizing: border-box;
  border-color: rgba(0, 0, 0, 0.1);
  padding: calc(4px * 4) calc(4px * 6);
}

.emotion-44>*+*:not( marquee ) {
  margin-left: calc(4px * 2);
}

.emotion-44>* {
  min-width: 0;
}

.emotion-44:first-of-type {
  border-top: none;
}

.emotion-44:first-of-type {
  border-top-left-radius: calc(2px - 1px);
  border-top-right-radius: calc(2px - 1px);
}

.emotion-44:last-of-type {
  border-bottom-left-radius: calc(2px - 1px);
  border-bottom-right-radius: calc(2px - 1px);
}

.emotion-46 {
  background: transparent;
  display: block;
  margin: 0!important;
  pointer-events: none;
  position: absolute;
  will-change: box-shadow;
  border-radius: inherit;
  bottom: 0;
  box-shadow: 0 0px 0px 0 rgba(0, 0, 0, 0);
  opacity: 1;
  left: 0;
  right: 0;
  top: 0;
  -webkit-transition: box-shadow 200ms cubic-bezier(0.08, 0.52, 0.52, 1);
  transition: box-shadow 200ms cubic-bezier(0.08, 0.52, 0.52, 1);
  border-radius: 2px;
}

@media ( prefers-reduced-motion: reduce ) {
  .emotion-46 {
    transition-duration: 0ms;
  }
}

<div>
  <div
    class=" woocommerce-payments-page"
    id=""
  >
    <div
      class="wcpay-banner-notice is-warning"
    >
      <div
        class="wcpay-banner-notice__content"
      >
        Viewing test loans. To view live loans, disable test mode in 
        <a
          href="admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments"
        >
          WooPayments settings
        </a>
        .
      </div>
    </div>
    <div
      class="components-surface components-card emotion-0 emotion-1"
      data-wp-c16t="true"
      data-wp-component="Card"
    >
      <div
        class="emotion-2 emotion-1"
      >
        <div
          class="components-flex components-card__header components-card-header wcpay-loan-summary-header emotion-4 emotion-1"
          data-wp-c16t="true"
          data-wp-component="CardHeader"
        >
          <div
            class="components-flex-item emotion-6 emotion-1"
            data-wp-c16t="true"
            data-wp-component="FlexItem"
          >
            Active loan overview
          </div>
          <div
            class="components-flex-item emotion-6 emotion-1"
            data-wp-c16t="true"
            data-wp-component="FlexItem"
          />
        </div>
        <div
          class="components-card__body components-card-body wcpay-loan-summary-body emotion-10 emotion-1"
          data-wp-c16t="true"
          data-wp-component="CardBody"
        >
          <div
            class="components-flex wcpay-loan-summary-row emotion-12 emotion-1"
            data-wp-c16t="true"
            data-wp-component="Flex"
          >
            <div
              class="components-flex-item components-flex-block wcpay-loan-summary-block emotion-14 emotion-1"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <div
                class="wcpay-loan-summary-block__title"
              >
                Total repaid
              </div>
              <div
                class="wcpay-loan-summary-block__value"
              >
                <span
                  class="is-big"
                >
                  $0.00
                </span>
                 of $1,100.00
              </div>
            </div>
            <div
              class="components-flex-item components-flex-block wcpay-loan-summary-block emotion-14 emotion-1"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <div
                class="wcpay-loan-summary-block__title"
              >
                Repaid this period (until Dec 27, 2024)
              </div>
              <div
                class="wcpay-loan-summary-block__value"
              >
                <span
                  class="is-big"
                >
                  $0.00
                </span>
                 of $122.23 minimum
              </div>
            </div>
          </div>
          <div
            class="components-flex wcpay-loan-summary-row is-bottom-row emotion-12 emotion-1"
            data-wp-c16t="true"
            data-wp-component="Flex"
          >
            <div
              class="components-flex-item components-flex-block wcpay-loan-summary-block emotion-14 emotion-1"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <div
                class="wcpay-loan-summary-block__title"
              >
                Loan disbursed
              </div>
              <div
                class="wcpay-loan-summary-block__value"
              >
                Oct 21, 2024
              </div>
            </div>
            <div
              class="components-flex-item components-flex-block wcpay-loan-summary-block emotion-14 emotion-1"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <div
                class="wcpay-loan-summary-block__title"
              >
                Loan amount
              </div>
              <div
                class="wcpay-loan-summary-block__value"
              >
                $1,000.00
              </div>
            </div>
            <div
              class="components-flex-item components-flex-block wcpay-loan-summary-block emotion-14 emotion-1"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <div
                class="wcpay-loan-summary-block__title"
              >
                Fixed fee
              </div>
              <div
                class="wcpay-loan-summary-block__value"
              >
                $100.00
              </div>
            </div>
            <div
              class="components-flex-item components-flex-block wcpay-loan-summary-block emotion-14 emotion-1"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <div
                class="wcpay-loan-summary-block__title"
              >
                Withhold rate
              </div>
              <div
                class="wcpay-loan-summary-block__value"
              >
                15
                %
              </div>
            </div>
            <div
              class="components-flex-item components-flex-block wcpay-loan-summary-block emotion-14 emotion-1"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <div
                class="wcpay-loan-summary-block__title"
              >
                First paydown
              </div>
              <div
                class="wcpay-loan-summary-block__value"
              >
                Oct 28, 2024
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="components-elevation emotion-30 emotion-1"
        data-wp-c16t="true"
        data-wp-component="Elevation"
      />
      <div
        aria-hidden="true"
        class="components-elevation emotion-30 emotion-1"
        data-wp-c16t="true"
        data-wp-component="Elevation"
      />
    </div>
    <div
      class="components-surface components-card woocommerce-table wcpay-loans-list emotion-34 emotion-35"
      data-wp-c16t="true"
      data-wp-component="Card"
    >
      <div
        class="emotion-36 emotion-35"
      >
        <div
          class="components-flex components-card__header components-card-header emotion-38 emotion-35"
          data-wp-c16t="true"
          data-wp-component="CardHeader"
        >
          <h2
            class="components-truncate components-text emotion-40 emotion-35"
            data-wp-c16t="true"
            data-wp-component="Text"
          >
            All loans
          </h2>
          <div
            class="woocommerce-table__actions"
          />
        </div>
        <div
          class="components-card__body components-card-body emotion-42 emotion-35"
          data-wp-c16t="true"
          data-wp-component="CardBody"
        >
          <div
            aria-labelledby="caption-0"
            class="woocommerce-table__table"
            role="group"
          >
            <table>
              <caption
                class="woocommerce-table__caption screen-reader-text"
                id="caption-0"
              >
                All loans
              </caption>
              <tbody>
                <tr>
                  <th
                    class="woocommerce-table__header is-left-aligned is-sorted"
                    role="columnheader"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      Disbursed
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      Disbursed
                    </span>
                  </th>
                  <th
                    class="woocommerce-table__header is-center-aligned is-left-aligned"
                    role="columnheader"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      Status
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      Status
                    </span>
                  </th>
                  <th
                    class="woocommerce-table__header is-numeric"
                    role="columnheader"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      Amount
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      Amount
                    </span>
                  </th>
                  <th
                    class="woocommerce-table__header is-numeric"
                    role="columnheader"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      Fixed fee
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      Fixed fee
                    </span>
                  </th>
                  <th
                    class="woocommerce-table__header is-numeric"
                    role="columnheader"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      Withhold rate
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      Withhold rate
                    </span>
                  </th>
                  <th
                    class="woocommerce-table__header is-numeric"
                    role="columnheader"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      First paydown
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      First paydown
                    </span>
                  </th>
                </tr>
                <tr>
                  <th
                    class="woocommerce-table__item is-left-aligned is-sorted"
                    scope="row"
                  >
                    <a
                      class="woocommerce-table__clickable-cell"
                      data-link-type="wc-admin"
                      href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&type=charge&filter=advanced&loan_id_is=financingoffer_1QEpQ2J5cIRIG92xWoVJJE3S"
                      tabindex="-1"
                    >
                      Oct 28, 2024
                    </a>
                  </th>
                  <td
                    class="woocommerce-table__item is-center-aligned is-left-aligned"
                  >
                    <a
                      class="woocommerce-table__clickable-cell"
                      data-link-type="wc-admin"
                      href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&type=charge&filter=advanced&loan_id_is=financingoffer_1QEpQ2J5cIRIG92xWoVJJE3S"
                      tabindex="-1"
                    >
                      <span
                        class="chip chip-warning"
                      >
                        Active
                      </span>
                    </a>
                  </td>
                  <td
                    class="woocommerce-table__item is-numeric"
                  >
                    <a
                      class="woocommerce-table__clickable-cell"
                      data-link-type="wc-admin"
                      href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&type=charge&filter=advanced&loan_id_is=financingoffer_1QEpQ2J5cIRIG92xWoVJJE3S"
                      tabindex="-1"
                    >
                      $1,000.00
                    </a>
                  </td>
                  <td
                    class="woocommerce-table__item is-numeric"
                  >
                    <a
                      class="woocommerce-table__clickable-cell"
                      data-link-type="wc-admin"
                      href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&type=charge&filter=advanced&loan_id_is=financingoffer_1QEpQ2J5cIRIG92xWoVJJE3S"
                      tabindex="-1"
                    >
                      $100.00
                    </a>
                  </td>
                  <td
                    class="woocommerce-table__item is-numeric"
                  >
                    <a
                      class="woocommerce-table__clickable-cell"
                      data-link-type="wc-admin"
                      href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&type=charge&filter=advanced&loan_id_is=financingoffer_1QEpQ2J5cIRIG92xWoVJJE3S"
                      tabindex="-1"
                    >
                      15%
                    </a>
                  </td>
                  <td
                    class="woocommerce-table__item is-numeric"
                  >
                    <a
                      class="woocommerce-table__clickable-cell"
                      data-link-type="wc-admin"
                      href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&type=charge&filter=advanced&loan_id_is=financingoffer_1QEpQ2J5cIRIG92xWoVJJE3S"
                      tabindex="-1"
                    >
                      -
                    </a>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="components-flex components-card__footer components-card-footer emotion-44 emotion-35"
          data-wp-c16t="true"
          data-wp-component="CardFooter"
        >
          <ul
            class="woocommerce-table__summary"
            role="complementary"
          >
            <li
              class="woocommerce-table__summary-item"
            >
              <span
                class="woocommerce-table__summary-value"
              >
                1
              </span>
              <span
                class="woocommerce-table__summary-label"
              >
                loan
              </span>
            </li>
            <li
              class="woocommerce-table__summary-item"
            >
              <span
                class="woocommerce-table__summary-value"
              >
                $1,000.00
              </span>
              <span
                class="woocommerce-table__summary-label"
              >
                total
              </span>
            </li>
            <li
              class="woocommerce-table__summary-item"
            >
              <span
                class="woocommerce-table__summary-value"
              >
                $100.00
              </span>
              <span
                class="woocommerce-table__summary-label"
              >
                fixed fees
              </span>
            </li>
          </ul>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="components-elevation emotion-46 emotion-35"
        data-wp-c16t="true"
        data-wp-component="Elevation"
      />
      <div
        aria-hidden="true"
        class="components-elevation emotion-46 emotion-35"
        data-wp-c16t="true"
        data-wp-component="Elevation"
      />
    </div>
  </div>
</div>
`;
